### DISCUSSION TITLE
[Question][Hw4] Do we need to implement cases of commands Read and Execute for capability too?

### DISCUSSION THREAD

**Discussion by ahnjonggun** [prosyslab-classroom/cs348-information-security#713] [Q&A]
Name: Jaewoo Ahn

YOUR QUESTION
Will simple read and execute commands without capability will be used in capability cases?
DO we need to implement read/execute both with and without the capabilities or only the read_with_capability and execute_with_capability?

**Comment by munimwasi** [comment]
For capability, only these 6 are necessary:

command := Create | Confer | Remove | Open | Read_with_capability | Execute_with_capability

**Comment by ahnjonggun** [comment]
Aha. Thank you.
Also, what happens if the name of the capability in the process does not match the name of the capability used when calling Execute_with_capability(first parameter of execute_with_capability)?
I wondered whether if we just return Access_denied or execute the next process by passing the capability of None.

**Comment by munimwasi** [comment]
I think there won't be any such case. However, if there are, it should be None.

**Comment by ahnjonggun** [comment]
Thank you!

**Comment by m-spitfire** [comment]
What do you mean there won't be any such case? Check #286
